### DISCUSSION TITLE
Use targeted carlini in UniversalPerturbation to generate noise but fail.

### DISCUSSION THREAD

**Discussion by MarcuXu** [Trusted-AI/adversarial-robustness-toolbox#2149] [Q&A]
**Describe the bug**
I hope to use the targeted carlini in UniversalPerturbation to generate noise, but the noise I get is just zeros arrays. Could you please help me with that?

**To Reproduce**
1. Clone the 1.14 ART and use the UniversalPerturbation function in my codes.
2. Set related parameters like this:
\# build targeted UAP
classifier = KerasClassifier(model=model)
\# print(type(model))
adv_crafter = UniversalPerturbation(
    classifier,
    attacker='carlini',
    delta=1e-2,
    attacker_params={'verbose': False, 'batch_size': 64, 
                    'targeted': True,
                     'confidence': 5e-2, 'learning_rate': 5e-2, 'max_iter': 10, 'initial_const': 5e-2},
    max_iter=20,
    verbose=True,
    \# eps=5.5,
    \# Only 1, 2 , `np.Inf` 1.1 and 1.2 supported for now.
    norm=np.Inf,)
\# set target label
target = 3
y_train_adv_tar = np.zeros(Y_train.shape)
for i in range(Y_train.shape[0]):
    y_train_adv_tar[i, target] = 1
\# generate noise
\_ = adv_crafter.generate(X_train, y=y_train_adv_tar)
noise = adv_crafter.noise
print(noise)
\# The upper codes work well when I use FGSM in TargetedUniversalPerturbation, so I think the logic of my codes could be right. Thus, I may forget or don't know how to generate the noise under carlini correctly.
**Expected behavior**
Get some noise which can be used to attack.

**Screenshots**

**System information (please complete the following information):**
 - Linux
 - Python version: 3.9
 - ART version or commit number: 1.14.0
 - TensorFlow / Keras / PyTorch / MXNet version: pip install . under the cloned ART.

**Comment by beat-buesser** [comment]
Hi @MarcuXu What is the output of your model, probabilities or logits?

I'll convert this issue into a discussion under the Discussions tab above to continue the discussion.

**Reply by MarcuXu** [reply]
Hi @beat-buesser, thank you for your reply! The inputs of training datasets are like x_train (`1000*200*52`) and y_train (`1000*6`), where 1000 is the sample number, `200*52` is the sample size, and 6 is the class number. Through the prediction from the CNN model (whose type is `<class 'tensorflow.python.keras.engine.sequential.Sequential'>`), like:
``` python
y = model.predict(x)
print(y.shape)
print(y[0])

# output:
# (200, 6)
# [9.9835736e-01 1.6425920e-03 6.0162335e-13 1.3994645e-16 1.0540235e-18 7.7685725e-09] 
# It looks like a one-hot array
#######
```
Could you please tell me how to implement the CW correctly in my CNN based on ART? I have been stuck in the problem for a week...
